katsdpimager/preprocess.cpp

```diff
@@ -334,19 +332,26 @@ void visibility_collector<P>::emit(int channel, vis_t<P> data[], std::size_t N)
 }
 
 template<int P>
-void visibility_collector<P>::compress(int channel)
+void visibility_collector<P>::compress(int channel, std::size_t buffer_size)
 {
-    if (buffer_size == 0)
-        return;  // some code will break on an empty buffer
     std::size_t out_pos = 0;
+    std::size_t i = 0;
+    // Skip initial flagged values
+    while (i < buffer_size && buffer[i].weights[0] == 0.0f)
+        i++;
```

Comment:
I moved skipping of flagged values from `add_impl2` to `compress` because the latter is serial, and filtering is harder to do in parallel code.

Reply:
I've always wondered what `impl2` means in that function name.